services/dynamic/dynamic-regex.service.js

```diff
@@ -0,0 +1,99 @@
+import Joi from 'joi'
+import RE2 from 're2'
+import { BaseService, InvalidParameter, queryParams } from '../index.js'
+import { url } from '../validators.js'
+import { renderDynamicBadge } from '../dynamic-common.js'
+
+export default class DynamicRegex extends BaseService {
+  static category = 'dynamic'
+  static route = {
+    base: `badge/dynamic/regex`,
+    pattern: '',
+    queryParamSchema: Joi.object({
+      url,
+      search: Joi.string().required(),
+      replace: Joi.string().optional(),
+      flags: Joi.string().optional(),
+      noMatch: Joi.string().optional(),
+    }),
+  }
+  static openApi = {
+    '/badge/dynamic/regex': {
+      get: {
+        summary: 'Dynamic Regex (re2) Badge [Experimental]',
```

Comment:
Lets make the title just "Dynamic Regex Badge" to keep the menu item concise. We can explain the rest of it in the documentation on the page